content/courses/advanced-ada/parts/resource_management/access_types.rst

```diff
@@ -15,29 +15,409 @@ Access types: Terminology
 
     Complete section!
 
+.. _Adv_Ada_Access_Types_Allocation:
 
 Access types: Allocation
 -------------------------
 
-.. admonition:: In the Ada Reference Manual
+Ada makes the distinction between pool-specific and general access types, as
+we'll discuss in this section. Before doing so, however, let's talk about
+memory allocation.
 
-    - :arm22:`3.10 Access Types <3-10>`
+In general terms, memory can be allocated dynamically on the
+heap or statically on the stack. For example:
 
-.. todo::
+.. code:: ada run_button project=Courses.Advanced_Ada.Resource_Management.Access_Types.Access_Types_Allocation.Simple_Allocation
 
-    Complete section!
+    procedure Show_Simple_Allocation is
+
+       --  Declaring access type:
+       type Integer_Access is access Integer;
 
+       --  Declaring access object:
+       A1 : Integer_Access;
 
-Pool-specific access types
---------------------------
+    begin
+       --  Allocating an Integer variable on the heap
```

Comment:
The only variable at that point is A1. Maybe "variable" should be "object"?

Reply:
Yes, that sounds good. I'll change it.